### PR TITLE
Push iotedgedev container to MCR

### PR DESCRIPTION
**Pull request by marianan**
MCR push process is already described at "MCR push process" OneNote. This change will make sure to not push to DockerHub but instead to MCR.

### CODE REVIEW

File: docker/tool/push-docker.sh [hunk 1]
```diff
@@ -1,59 +1,60 @@
...
-echo -e "\n===== Creating Multi-Arch Docker image"
-#docker manifest create $IMAGE_NAME:latest $IMAGE_NAME:latest-amd64 $IMAGE_NAME:latest-windows-amd64 
-docker manifest create --insecure $IMAGE_NAME:latest $IMAGE_NAME:latest-amd64 
-
-echo -e "\n===== Pushing Docker Multi-Arch image"
-docker manifest push --purge $IMAGE_NAME:latest
-
-if [ in_docker_folder = 0 ]; then
-    cd original_folder
+#!/bin/bash
```
Comment (mhshami01) [inline docker/tool/push-docker.sh:1]: did we change line endings or something? was it intentional?

**Review by avagraw — APPROVED** [review]

File: docker/tool/push-docker.sh (hunk 1 above)
Comment (marianan) [reply docker/tool/push-docker.sh:1]: there is no diff in this line, will connect offline to udnerstand your question.

**Comment by cindydeng1998** [comment]
It doesn't look like this file has been edited in 3 years, and we have manual steps in OneNote to push images. Is the process now to use this script instead of the OneNote documentation?

File: docker/tool/push-docker.sh (hunk 1 above)
Comment (mhshami01) [reply docker/tool/push-docker.sh:1]: on master:
image
on your branch:
image

Comment (mhshami01) [reply docker/tool/push-docker.sh:1]: as you can see, on branch user/marine/mcrpush, the file is now using windows style line endings, which not good in a bash script :-)

**Comment by marianan** [comment]
i think we should use the script and make sure that it works for what we need it for. We already have some many places with instructions in documentation.

---
In reply to: [858800036](https://github.com/Azure/iotedgedev/pull/478#issuecomment-858800036) [](http://example.com/codeflow?ancestors=858800036,858800036)

File: docker/tool/push-docker.sh (hunk 1 above)
Comment (marianan) [reply docker/tool/push-docker.sh:1]: good catch, i will update

File: docker/tool/push-docker.sh [hunk 2]
```diff
@@ -1,59 +1,60 @@
...
+    in_docker_folder=1
+else
+    in_docker_folder=0
+    cd docker
+fi
+
+function show_help
+{
+    echo "Usage:"
+    echo "push-docker.sh <dockerhub> <platform> [<version>]"
```
Comment (cindydeng1998) [inline docker/tool/push-docker.sh:19]: the first entry here is dockerhub, but the first argument is IMAGE_NAME="$1" on line 27. which one should we keep?

File: docker/tool/push-docker.sh [hunk 3]
```diff
@@ -1,59 +1,60 @@
...
+#!/bin/bash
+
+# stop on error
+set -e
+
+# make sure we're in docker folder
+original_folder=$PWD
+
+if [ -z $"echo $PWD | grep /docker$" ]; then 
+    in_docker_folder=1
```
[the remark this answers is not in the document]
Comment (marianan) [reply docker/tool/push-docker.sh:10]: I will remove this logic here and at the end of the file

File: docker/tool/push-docker.sh [hunk 4]
```diff
@@ -1,59 +1,60 @@
...
+    echo "push-docker.sh imagename [linux|windows]"
+    exit 1
+fi
+
+if [ -z "$VERSION" ]; then
+    echo -e "\n===== Detecting version"
+    VERSION=$(cat ../iotedgedev/__init__.py | grep '__version__' | grep -oP "'\K[^']+")
+    echo "Detected version $VERSION"
+fi
+
```
[the remark this answers is not in the document]
Comment (marianan) [reply docker/tool/push-docker.sh:43]: will push these 2 containers as well:
-t $IMAGE_NAME:$VERSION \
        -t $IMAGE_NAME:latest \

File: docker/tool/push-docker.sh [hunk 5]
```diff
@@ -1,60 +1,35 @@
...
+IMAGE_NAME="iotedgedev"
+VERSION="$1"
+
+if [ "$VERSION" = "--help" ]; then    
+    show_help
+fi
+
+if [ -z "$VERSION" ]; then
+    echo -e "\n===== Detecting version"
+    VERSION=$(cat ../../iotedgedev/__init__.py | grep '__version__' | grep -oP "'\K[^']+")
```
Comment (mhshami01) [inline docker/tool/push-docker.sh:27]: Could be simplified as follows...
VERSION=$(grep '__version__' ../../iotedgedev/__init__.py  | grep -oP "'\K[^']+")

**Review by mhshami01 — APPROVED** [review]

**Review by cindydeng1998 — APPROVED** [review]
🌞

**Review by mhshami01 — APPROVED** [review]

**Pull request merged**